### DISCUSSION TITLE
Adding HostedService to module Startup

### DISCUSSION THREAD

**Discussion by adamico2412** [OrchardCMS/OrchardCore#14380] [Q&A] [answered]
Is it possible to add an IHostedService inside a module's Startup class?  After attempting to do so, the hosted service never runs.  I know there are background tasks, but I need a service that runs continuously in the background.  Thanks for any help in advance.

**Comment by lampersky** [comment, marked as answer]
@adamico2412 HostedService is not supported inside modules.
You can read more about possible workarounds here: https://github.com/OrchardCMS/OrchardCore/issues/8409#issuecomment-831963537
